tarsier/core.py

```diff
@@ -1,20 +1,29 @@
 from asyncio import Protocol
 from pathlib import Path
-from typing import Dict, Tuple
+from typing import Tuple, Optional, TypedDict
 
 from tarsier._utils import load_js
 from tarsier.adapter import AnyDriver, BrowserAdapter, adapter_factory
 from tarsier.ocr import OCRService
 from tarsier.text_format import format_text
 
-TagToXPath = Dict[int, str]
+
+class TagMetadata(TypedDict):
+    tarsier_id: int
+    element_name: str
+    opening_tag_html: str
+    xpath: str
+    element_text: Optional[str]
+    text_node_index: Optional[int]
+    id_symbol: str
+    id_string: str
 
 
 class ITarsier(Protocol):
-    async def page_to_image(self, driver: AnyDriver) -> Tuple[bytes, Dict[int, str]]:
+    async def page_to_image(self, driver: AnyDriver) -> Tuple[bytes, list[TagMetadata]]:
         raise NotImplementedError()
 
-    async def page_to_text(self, driver: AnyDriver) -> Tuple[str, Dict[int, str]]:
+    async def page_to_text(self, driver: AnyDriver) -> Tuple[str, list[TagMetadata]]:
         raise NotImplementedError()
```

Comment:
This is missing `page_to_image_and_text`. We should add it either here or in a different PR